From c90f4b4880e3c243dc3113d85613152b70ccccf1 Mon Sep 17 00:00:00 2001
From: Zhang Fuxin <fxzhang@ict.ac.cn>
Date: Wed, 24 Apr 2019 16:44:35 +0800
Subject: [PATCH 1/4] add option     SYS_SUPPORTS_HOTPLUG_CPU    
 ARCH_SUSPEND_POSSIBLE     SUSPEND to enable the selection of PM_SLEEP etc.
 Add play_dead() for loongson2. They are needed by android healthd(epoll
 syscall).

---
 arch/mips/loongson2/Kconfig | 3 +++
 arch/mips/loongson2/smp.c   | 8 ++++++++
 2 files changed, 11 insertions(+)

diff --git a/arch/mips/loongson2/Kconfig b/arch/mips/loongson2/Kconfig
index d0f0d0d..3229954 100644
--- a/arch/mips/loongson2/Kconfig
+++ b/arch/mips/loongson2/Kconfig
@@ -21,6 +21,9 @@ config 2K1000_SBC
 	select ARCH_SUPPORTS_MSI
 	select USE_OF
 	select ZONE_DMA32 if SWIOTLB
+	select SYS_SUPPORTS_HOTPLUG_CPU
+	select ARCH_SUSPEND_POSSIBLE
+	select SUSPEND
 	help
 		Loongson 2K1000 SBC
 endchoice
diff --git a/arch/mips/loongson2/smp.c b/arch/mips/loongson2/smp.c
index c8f924e..ad14fd4 100644
--- a/arch/mips/loongson2/smp.c
+++ b/arch/mips/loongson2/smp.c
@@ -29,6 +29,14 @@
 #include <asm/mach-loongson2/2k1000.h>
 #include <asm/smp.h>
 
+#ifdef CONFIG_SYS_SUPPORTS_HOTPLUG_CPU
+/* Todo */
+void play_dead(void)
+{
+    printk("dead!\n");
+    while(1) {};
+}
+#endif
 
 DEFINE_PER_CPU(int, cpu_state);
 static uint32_t core0_c0count[NR_CPUS];
-- 
2.7.4

